src/mscorlib/src/System/Environment.cs

```diff
@@ -703,6 +693,9 @@ private static void SetEnvironmentVariableCore(string variable, string value)
                         // The error message from Win32 is "The filename or extension is too long",
                         // which is not accurate.
                         throw new ArgumentException(SR.Format(SR.Argument_LongEnvVarValue));
+                    case Win32Native.ERROR_NOT_ENOUGH_MEMORY:
```

Comment:
I guess Unix should give the same result
https://github.com/dotnet/coreclr/blob/master/src/pal/src/misc/environ.cpp#L345